examples/ct/ct_handlers.go

```diff
@@ -251,9 +251,9 @@ func addChainInternal(w http.ResponseWriter, r *http.Request, c RequestHandlers,
 
 	request := trillian.QueueLeavesRequest{LogId: c.logID, Leaves: []*trillian.LeafProto{&leafProto}}
 
-	ctx, _ := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
-
+	ctx, cancelFunc := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
 	response, err := c.rpcClient.QueueLeaves(ctx, &request)
+	cancelFunc()
```

Comment:
The docs say you should call cancel as soon as the work is complete but deferring it guarantees it gets called regardless of what happens afterwards. I think I'll go with defer.